sync-core/src/main/java/com/cloudant/sync/query/IndexCreator.java

```diff
@@ -0,0 +1,291 @@
+                    parameters.put("index_name", indexName);
+                    parameters.put("index_type", indexType);
+                    parameters.put("field_name", fieldName);
+                    parameters.put("last_sequence", 0);
+                    long rowId = database.insert(IndexManager.INDEX_METADATA_TABLE_NAME,
+                                                 parameters);
+                    if (rowId < 0) {
+                        transactionSuccess = false;
+                        break;
+                    }
+                }
+
+                String statement = null;
+                try {
+                    List<String> columnList = new ArrayList<String>();
+                    for (String field: fieldNamesList) {
+                        columnList.add("\"" + field + "\"");
+                    }
+                    // Create the table for the index
+                    statement = createIndexTableStatementForIndexName(indexName, columnList);
+                    database.execSQL(statement);
+
+                    // Create the SQLite index on the index table
+                    statement = createIndexIndexStatementForIndexName(indexName, columnList);
+                    database.execSQL(statement);
+                } catch (SQLException e) {
+                    String msg = String.format("Index creation error occurred (%s):",statement);
+                    logger.log(Level.SEVERE, msg, e);
+                    transactionSuccess = false;
+                }
+
+                if (transactionSuccess) {
+                    database.setTransactionSuccessful();
+                }
+                database.endTransaction();
+
+                return transactionSuccess;
+            }
+        });
+
+        // Update the new index if it's been created
+        boolean success;
+        try {
+            success = result.get();
+        } catch (ExecutionException e) {
+            logger.log(Level.SEVERE, "Execution error encountered:", e);
+            return null;
+        } catch (InterruptedException e) {
+            logger.log(Level.SEVERE, "Execution interrupted error encountered:", e);
+            return null;
+        }
+
+        if (success) {
+            success = IndexUpdater.updateIndex(indexName,
+                                               fieldNamesList,
+                                               database,
+                                               datastore,
+                                               queue);
+        }
+
+        return success ? indexName : null;
+    }
+
+    /**
+     *  Validate the field name string is usable.
+     *
+     *  The only restriction so far is that the parts don't start with
+     *  a $ sign, as this makes the query language ambiguous.
+     */
+    protected static boolean validFieldName(String fieldName) {
+        String[] parts = fieldName.contains(".") ? fieldName.split("\\.") : new String[]{fieldName};
```

Comment:
Doesn't look like you need a `contains` check here, `"mike".split("\\.") ==> [mike]`